pkg/controller/direct/directbase/directbase_controller.go

```diff
@@ -53,22 +54,41 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/source"
 )
 
-// Add creates a new controller for reconciling objects of the specified GVK, delegating actual resource reconciliation to the provided Model.
-func Add(mgr manager.Manager, gvk schema.GroupVersionKind, model Model, opts Deps) error {
+var ControllerBuilder directControllerBuilder
+
+func init() {
+	ControllerBuilder = directControllerBuilder{modelMapper: map[schema.GroupVersionKind]func(*controller.Config) Model{}}
```

Comment:
Nit: We may be able to get away without an init function here.  But ... I can see it getting more complicated, so only a nit!

Reply:
Done.